saver.go

```diff
@@ -317,6 +340,63 @@ func (s JSONFilesHistorySaver) SaveRelatedChats(chats []mtproto.TL) error {
 	return nil
 }
 
+func (s JSONFilesHistorySaver) SaveContacts(contacts []mtproto.TL) error {
+
+	var encoder *json.Encoder
+
+	if encoder == nil {
```

Comment:
It's always `nil`, so the check is unnecessary. It could be just `encoder := json.NewEncoder(file)`.